src/lib.rs

```diff
@@ -562,27 +561,35 @@ fn execute_instr(input: InstrContext) -> Option<InstrEffects> {
     program_cache.environments = environments.clone();
     program_cache.upcoming_environments = Some(environments);
 
-    let tx_batch_processor = TransactionBatchProcessor::<DummyForkGraph>::new(
-        clock.slot,
-        clock.epoch,
-        epoch_schedule,
-        FeeStructure::default(),
-        Arc::<RuntimeConfig>::default(),
-        Arc::new(RwLock::new(program_cache)),
-        loaded_builtins,
-    );
+    // let tx_batch_processor = TransactionBatchProcessor::<DummyForkGraph>::new(
+    //     clock.slot,
+    //     clock.epoch,
+    //     epoch_schedule,
+    //     Arc::<RuntimeConfig>::default(),
+    //     Arc::new(RwLock::new(program_cache)),
+    //     loaded_builtins,
+    // );
 
     for acc in &input.accounts {
         if acc.1.executable && programs_loaded_for_tx_batch.find(&acc.0).is_none() {
+            /* pub fn load_program_with_pubkey<CB: TransactionProcessingCallback, FG: ForkGraph>(
+    callbacks: &CB,
+    program_cache: &ProgramCache<FG>,
+    pubkey: &Pubkey,
+    slot: Slot,
+    effective_epoch: Epoch,
+    epoch_schedule: &EpochSchedule,
+    reload: bool,
+) -> Option<Arc<ProgramCacheEntry>> { */
             if let Some(loaded_program) =
```

Comment:
Great to see comments! :D

One approach thing we've found which works really well is putting GitHub links to the Agave source code, to explain where logic comes from. See example: https://github.com/firedancer-io/firedancer/blob/vm-belt-sanding-public/src/flamenco/vm/syscall/fd_vm_syscall_hash.c